### PR TITLE
Added README

### PR DESCRIPTION
**Pull request by mannol**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# APIDSL
+A [DSL](http://en.wikipedia.org/wiki/Domain-specific_language) for **C** API's to add consistency of naming schemes and comments.
+
+## INSTALATION
```
Comment (iphydf) [inline README.md]: Installation (double L and lowercase).

File: README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+You will need **OCaml** (>= 4.0; older versions may not work) and **OPAM** (>=1.2). Instructions on how to install **OCaml** and **OPAM** can be found [here](https://opam.ocaml.org/doc/Install.html).
```
Comment (iphydf) [inline README.md]: - 4.02
- add space between ">=" and "1.2"

File: README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+### Installing dependencies:
+Make sure to configure **OPAM** by running ``opam init``. (Depending on your configuration you might also need to run ``eval `opam config env` ``).
+Running ``opam install ocamlfind ppx_deriving menhir`` should install all dependencies required for APIDSL.
+
+### Compiling
+Just run ``make`` in APIDSL's root directory. The apidsl binary can be found under ``./_build/apigen.native``
```
Comment (iphydf) [inline README.md]: APIDSL or apidsl?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+## USAGE
+TODO but, you can check some examples in tests directory.
```
Comment (iphydf) [inline README.md]: Newline at EOF. Move comma to before "but", i.e. "TODO, but you ..."

File: README.md (hunk 3 above)
Comment (mannol) [inline README.md]: Former as it's abbreviation. Makes more sense IMHO.

**Pull request merged**